core/src/apps/cardano/address.py

```diff
@@ -43,19 +46,83 @@ def validate_address_parameters(parameters: CardanoAddressParametersType) -> Non
     if parameters.address_type == CardanoAddressType.BYRON:
         if not is_byron_path(parameters.address_n):
             raise INVALID_ADDRESS_PARAMETERS
-    elif parameters.address_type in ADDRESS_TYPES_SHELLEY:
-        if not is_shelley_path(parameters.address_n):
-            raise INVALID_ADDRESS_PARAMETERS
 
+    elif parameters.address_type in ADDRESS_TYPES_SHELLEY:
         if parameters.address_type == CardanoAddressType.BASE:
```

Comment:
I'd rather leave it as is mainly for backwards compatibility reasons. If I rename it here, I would IMHO have to rename it also in the outwards facing API in Connect for it to make sense. This would break existing client code, which I don't want - but maybe that's negligible.